crates/router/src/core/payment_methods.rs

```diff
@@ -2650,3 +2667,169 @@ impl pm_types::SavedPMLPaymentsInfo {
         Ok(())
     }
 }
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+#[allow(clippy::too_many_arguments)]
+async fn add_token_call_to_store(
+    state: SessionState,
+    req_state: routes::app::ReqState,
+    merchant_account: domain::MerchantAccount,
+    profile: domain::Profile,
+    key_store: domain::MerchantKeyStore,
+    payment_method_create_request: &payment_methods::PaymentMethodCreate,
+    payment_method: &api::PaymentMethodResponse,
+    payment_method_session: &domain::payment_methods::PaymentMethodSession,
+) -> RouterResult<()> {
+    use crate::{db::errors::ConnectorErrorExt, types::Tokenizable};
+
+    let customer_id = payment_method_create_request.customer_id.to_owned();
+    let connector_id = payment_method_create_request
```

Comment:
make a impl based function `get_tokenize_connector_id` on the `payment_method_create_request` type and move the logic there